### ISSUE TITLE
Video filter still active after closing content

### ISSUE THREAD

**Issue by Denshadego** [libretro/RetroArch#12622]
## Description

Any video filter chosen after loading content is still active after closing content.

### Expected behavior

Video filters should work the same as any setting. If you select one (like any of the Blaarg filters) after loading content, the specific filter should only be remembered by the core, if overrides are saved for that core, and the setting should be back to the state it was before loading content.

Example: If no filter was selected when opening retroarch, and then loading content and selecting a filter, no filter should be active after closing content. Only remembered by the core if core/game overrides were saved.

### Actual behavior

After loading content with no video filter previously loaded, and choosing any video filter in video settings. When closing content, the video filter is still active, instead of rolling back to the default state of no video filter selected.

The filter is even active when loading different content in other cores with no previously filter related override saved.

### Steps to reproduce the bug

1. Open Retroarch with default setting (no video filter active).
2. Load content.
3. With said content still loaded, pick a video filter in video settings (optionally saving core overrides) 
4. Close content.
5. Video filter is still selected.
6. Load different content.
7. Same video filter still selected.

Retroarch version: Latest stable 1.96
System:

**Comment by hizzlekizzle** [comment]
I was not able to reproduce any unexpected behavior in linux 64-bit. That is, if I load the filter with content running via a core that has no overrides applied, the video filter remains selected when I move to another core. If I load it with content running via a core that has an active override, it does not remain selected when I move to another core.

If I save the video filter to an override, it loads correctly when the core is loaded and unloads correctly when I switch to another core.

**Comment by Denshadego** [comment]
You're right, mi bad! Did further testing and this is only happening in Android. On windows, when closing content with a filter loaded, it goes back to no filter selected as default.
